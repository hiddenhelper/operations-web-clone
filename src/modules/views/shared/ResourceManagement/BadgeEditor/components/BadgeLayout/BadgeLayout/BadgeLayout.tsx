import React, { memo, useMemo } from 'react';

import Avatar from '@material-ui/core/Avatar';
import Badge from '@material-ui/core/Badge';
import PersonIcon from '@material-ui/icons/Person';
import Typography from '@material-ui/core/Typography';

import { BadgeModel } from '../../../../../../../models';
import { BarcodeIcon } from '../../../../../../../../constants';
import { getConditionalDefaultValue, getDefaultValue, formatBadgeCode } from '../../../../../../../../utils/generalUtils';
import { useStyles, BadgeLayoutStyles } from '../../../../BadgeTemplate/styles';
import { avatarGlobalStyles } from '../../../../../../../../assets/styles/Avatars/styles';

export interface IBadgeLayout {
  layoutClassSelector: any;
  nameWithBackground?: string;
  nameColor?: string;
  avatarPosition?: string;
  avatarBorder?: string;
  toggleClass: string;
  showLogo?: boolean;
  currentBgColor: string;
  currentFont: string;
  currentFontColor: string;
  logo?: string;
  badge: BadgeModel.IBadge;
}

const BadgeLayout = ({
  currentBgColor,
  currentFont,
  currentFontColor,
  layoutClassSelector,
  nameWithBackground = '',
  nameColor = '',
  avatarPosition = '',
  avatarBorder = '',
  toggleClass,
  showLogo = false,
  logo,
  badge,
}: IBadgeLayout) => {
  const classes = useStyles();
  const avatarGlobalClasses = avatarGlobalStyles();
  const generalLayoutClasses = BadgeLayoutStyles();
  const logoToShow = useMemo(() => logo || badge?.badgeTemplate?.logoUrl, [logo, badge]);

  return (
    <>
      <div
        style={{ backgroundColor: `#${currentBgColor}` }}
        className={`frontView ${layoutClassSelector.dataContainerSecondaryBg} ${classes.dataContainerOverlap} ${generalLayoutClasses.badgeGeneralRadius} ${toggleClass}`}
      >
        <div className={`${layoutClassSelector.container} ${classes.dataContainer} ${generalLayoutClasses.badgeInfoRadius}`}>
          {showLogo && !logoToShow && (
            <div className={`${layoutClassSelector.imageWrapper} ${generalLayoutClasses.placeholderWrapper} ${layoutClassSelector.placeholderColor}`}>
              Badge Logo
            </div>
          )}
          {showLogo && logoToShow && <img className={layoutClassSelector.imageWrapper} srcSet={`${logoToShow} 1x, ${logoToShow} 2x`} alt="Badge Logo" />}
          <div className={avatarPosition}>
            <Badge
              className={`${classes.badgePosition} ${avatarGlobalClasses.badgeMedium}`}
              overlap="circle"
              anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'right',
              }}
            >
              <Avatar className={` ${avatarGlobalClasses.avatarWrapper} ${avatarGlobalClasses.badgeMedium} ${avatarBorder}`}>
                {getConditionalDefaultValue(
                  badge?.pictureUrl,
                  <img alt="" style={{ width: '100%' }} src={badge?.pictureUrl} />,
                  <PersonIcon titleAccess={''} />
                )}
              </Avatar>
            </Badge>
          </div>
          <div
            style={{ backgroundColor: `#${nameColor}` }}
            className={`${nameWithBackground} ${layoutClassSelector.badgeTextPosition} ${classes.badgeTextWrapper}`}
          >
            <Typography style={{ fontFamily: currentFont, color: `#${currentFontColor}` }} className={generalLayoutClasses.badgeLayoutTitle}>
              {getDefaultValue(badge?.firstName, 'Worker')} {getDefaultValue(badge?.lastName, 'Name')}
            </Typography>
            <Typography style={{ fontFamily: currentFont, color: `#${currentFontColor}` }} className={generalLayoutClasses.badgeLayoutSubtitle}>
              {getDefaultValue(badge?.companyName, 'Company Name')}
            </Typography>
          </div>
          <div className={`${layoutClassSelector.subDataContainer} ${classes.badgeTextWrapper}`}>
            <Typography style={{ fontFamily: currentFont }} className={`${generalLayoutClasses.badgeLayoutSubtitle} ${generalLayoutClasses.badgeCode}`}>
              {getConditionalDefaultValue(!!badge?.code, formatBadgeCode(getDefaultValue(badge?.code, '')), 'XX 00 XX')}
            </Typography>
          </div>
          <div className={`${classes.barcodePosition}`}>
            <BarcodeIcon />
          </div>
        </div>
      </div>
    </>
  );
};

export default memo(BadgeLayout);
