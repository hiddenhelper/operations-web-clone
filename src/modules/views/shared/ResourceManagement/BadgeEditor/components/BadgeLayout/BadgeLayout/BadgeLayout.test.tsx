import React from 'react';
import { act, fireEvent, render, RenderResult } from '@testing-library/react';

import BadgeLayout, { IBadgeLayout } from './BadgeLayout';

describe.skip('BadgeEditor', () => {
  let wrapper: RenderResult;
  let props: IBadgeLayout;

  beforeEach(() => {
    props = {
      currentBgColor: '',
      currentFont: '',
      currentFontColor: '',
      layoutClassSelector: '',
      toggleClass: '',
    };
  });

  it('should render', () => {
    wrapper = render(<BadgeLayout {...props} />);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render logo', () => {
    props.logo = 'logo';
    props.showLogo = true;
    wrapper = render(<BadgeLayout {...props} />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
