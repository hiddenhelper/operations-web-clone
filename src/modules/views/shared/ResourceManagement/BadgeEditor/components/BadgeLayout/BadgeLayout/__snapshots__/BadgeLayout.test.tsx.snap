// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeEditor should render 1`] = `
<div>
  <div
    class="frontView undefined makeStyles-dataContainerOverlap-7 makeStyles-badgeGeneralRadius-59 "
  >
    <div
      class="undefined makeStyles-dataContainer-6 makeStyles-badgeInfoRadius-60"
    >
      <div
        class=""
      >
        <span
          class="MuiBadge-root makeStyles-badgePosition-15 makeStyles-badgeMedium-44"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle  makeStyles-avatarWrapper-33 makeStyles-badgeMedium-44  MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <span
            class="MuiBadge-badge MuiBadge-anchorOriginBottomRightCircle MuiBadge-invisible"
          />
        </span>
      </div>
      <div
        class=" undefined makeStyles-badgeTextWrapper-11"
      >
        <p
          class="MuiTypography-root makeStyles-badgeLayoutTitle-51 MuiTypography-body1"
        >
          Worker
           
          Name
        </p>
        <p
          class="MuiTypography-root makeStyles-badgeLayoutSubtitle-52 MuiTypography-body1"
        >
          Company Name
        </p>
      </div>
      <div
        class="undefined makeStyles-badgeTextWrapper-11"
      >
        <p
          class="MuiTypography-root makeStyles-badgeLayoutSubtitle-52 makeStyles-badgeCode-49 MuiTypography-body1"
        >
          XX 00 XX
        </p>
      </div>
      <div
        class="makeStyles-barcodePosition-17"
      >
        <svg>
          barcode.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`BadgeEditor should render logo 1`] = `
<div>
  <div
    class="frontView undefined makeStyles-dataContainerOverlap-73 makeStyles-badgeGeneralRadius-125 "
  >
    <div
      class="undefined makeStyles-dataContainer-72 makeStyles-badgeInfoRadius-126"
    >
      <img
        alt="Badge Logo"
        srcset="logo 1x, logo 2x"
      />
      <div
        class=""
      >
        <span
          class="MuiBadge-root makeStyles-badgePosition-81 makeStyles-badgeMedium-110"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle  makeStyles-avatarWrapper-99 makeStyles-badgeMedium-110  MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <span
            class="MuiBadge-badge MuiBadge-anchorOriginBottomRightCircle MuiBadge-invisible"
          />
        </span>
      </div>
      <div
        class=" undefined makeStyles-badgeTextWrapper-77"
      >
        <p
          class="MuiTypography-root makeStyles-badgeLayoutTitle-117 MuiTypography-body1"
        >
          Worker
           
          Name
        </p>
        <p
          class="MuiTypography-root makeStyles-badgeLayoutSubtitle-118 MuiTypography-body1"
        >
          Company Name
        </p>
      </div>
      <div
        class="undefined makeStyles-badgeTextWrapper-77"
      >
        <p
          class="MuiTypography-root makeStyles-badgeLayoutSubtitle-118 makeStyles-badgeCode-115 MuiTypography-body1"
        >
          XX 00 XX
        </p>
      </div>
      <div
        class="makeStyles-barcodePosition-83"
      >
        <svg>
          barcode.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;
